deployment/Tiltfile

```diff
@@ -0,0 +1,124 @@
+def deployment_create(name, image, command=None, namespace="", replicas=None, ports=[]):
+    """
+    Create a Kubernetes deployment in the current cluster. If ports specified, 
+    create a Kubernetes service connected to the given port(s) in the deployment.
+
+    Args:
+      name: The deployment name
+      image: The image name
+      command: The command to run, if different from the entrypoint in the image
+      namespace: The namespace to create the deployment in, if different from the current namespace.
+      ports: The ports to expose as a ClusterIP service.
+    """
+    port = None
+
+    if type(ports) == "string":
+        ports = [ports]
+
+    if len(ports) > 0:
+        port = ports[0].split(':')[-1]
+
+    blobs = [deployment_yaml(name, image, command, namespace, replicas, port)]
+
+    if len(ports) > 0:
+        blobs.append(service_yaml(name, namespace=namespace, ports=ports))
+
+    k8s_yaml(blobs)
+
+
+def deployment_yaml(name, image, command=None, namespace="", replicas=None, port=None):
+    """
+    Create Kubernetes YAML for a simple deployment.
+
+    Args:
+      name(string): The deployment name
+      image(string): The image name
+      command(string or list): The command to run, if different from the entrypoint in the image
+      namespace(string): The namespace to create the deployment in, if different from the current namespace.
+      replicas(int): The number of replicas to create if different than 1.
+      port(string): The container port to declare
+
+    Returns:
+      The deployment YAML as a blob
+    """
+
+    args = [
+        "kubectl",
+        "create",
+        "deployment",
+        name,
+    ]
+
+    args.extend(create_args(image, command, namespace, replicas=replicas, port=port))
+    return local(args, quiet=True)
+
+
+valid_svc_types = ['clusterip', 'nodeport', 'loadbalancer', 'externalname']
+
+
+def service_yaml(name, svc_type='ClusterIP', external_name=None, namespace="", ports=[]):
+    """
+    Create Kubernetes YAML for a service.
+
+    Args:
+      name(string): The service name
+      svc_type(string): The service type (default ClusterIP)
+      external_name(string): The external name to use with svc_type=ExternalName
+      namespace(string): The namespace to create the service in, if different from the current namespace.
+      ports(string or list[string]): The port pairs to use for the service
+    """
+    svc_type = svc_type.lower()
+    if not svc_type in valid_svc_types:
```

Comment:
seems odd to do custom validation here, rather than using only the kubectl validation

Reply:
I didn't check what kubectl will say, is it helpful?

Reply:
```
$ kubectl create service x
error: unknown command "x"
See 'kubectl create service -h' for help and examples
```

:grimacing:

Reply:
It's a little misleading without the validation. 
With:
```
Initial Build • (Tiltfile)
Loading Tiltfile at: /Users/nicksieger/tilt-dev/tilt-extensions/deployment/test/Tiltfile
local: kubectl create deployment html --image=deployment_html --port=80 --dry-run=client -o=yaml
local: kubectl create service clusterip html --tcp=80 --dry-run=client -o=yaml
ERROR: Traceback (most recent call last):
  /Users/nicksieger/tilt-dev/tilt-extensions/deployment/test/Tiltfile:10:19: in <toplevel>
  /Users/nicksieger/tilt-dev/tilt-extensions/deployment/Tiltfile:72:13: in service_yaml
  <builtin>: in fail
Error: svc_type badtype not recognized, must be one of: ["clusterip", "nodeport", "loadbalancer", "externalname"]
```
Without:
```
Initial Build • (Tiltfile)
Loading Tiltfile at: /Users/nicksieger/tilt-dev/tilt-extensions/deployment/test/Tiltfile
local: kubectl create deployment html --image=deployment_html --port=80 --dry-run=client -o=yaml
local: kubectl create service clusterip html --tcp=80 --dry-run=client -o=yaml
local: kubectl create service badtype html --tcp=443 --dry-run=client -o=yaml
ERROR: Traceback (most recent call last):
  /Users/nicksieger/tilt-dev/tilt-extensions/deployment/test/Tiltfile:10:19: in <toplevel>
  /Users/nicksieger/tilt-dev/tilt-extensions/deployment/Tiltfile:86:17: in service_yaml
  <builtin>: in local
Error: command "kubectl create service badtype html --tcp=443 --dry-run=client -o=yaml" failed.
error: exit status 1
stdout: ""
stderr: "Error: unknown flag: --tcp\nSee 'kubectl create service --help' for usage.\n"
```
Is your point that it limits use of future service types?